unimod_mapper/unimod_mapper.py

```diff
@@ -1248,13 +1292,17 @@ def map_mods(self, mod_list):
             else:
                 unimod_name = mod_dict["name"]
                 chemical_formula = mod_dict["composition"]
+                # composition_unimod_style = chemical_formula
+                cc_string = "+" + "".join(
+                    ["{0}:{1}".format(k, v) for k, v in mod_dict["composition"].items()]
+                )
                 chemical_composition = ChemicalComposition()
-                chemical_composition.use(formula=chemical_formula)
+                chemical_composition.use(formula=cc_string)
                 composition = chemical_composition
                 composition_unimod_style = chemical_composition.hill_notation_unimod()
-                unimod_name_list = self.composition_to_name(composition_unimod_style)
-                unimod_id_list = self.composition_to_id(composition_unimod_style)
-                mass = self.composition_to_mass(composition_unimod_style)
+                unimod_name_list = self.composition_to_names(chemical_formula)
+                unimod_id_list = self.composition_to_ids(chemical_formula)
+                mass = self.composition_to_mass(chemical_formula)
```

Comment:
I'm pretty confused about this whole part.

This looks like `mod_dict["composition"]` should be a dict, which is not how it's defined in the doc string (`"composition": None, # For user-defined mods composition needs to be given as a Hill notation`) though I would support switching to dict.

But then the cc_string formatting does not look correct. ChemicalComposition uses formulas in "hill notation (or unimod hill notation)", so shouldn't it be `["{0}({1})".format(k, v) for k, v in mod_dict["composition"].items()]`?

Also line 1302 (composition_unimod_style) can be removed since it's not used anywhere.